components/style/values/computed/image.rs

```diff
@@ -0,0 +1,291 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! CSS handling for the computed value of
+//! [`image`][image]s
+//!
+//! [image]: https://drafts.csswg.org/css-images/#image-values
+
+use cssparser::Color as CSSColor;
+use std::fmt;
+use std::fmt::Debug;
+use url::Url;
+use values::LocalToCss;
+use values::computed::{Context, Length, LengthOrPercentage, ToComputedValue};
+use values::specified;
+use values::specified::UrlExtraData;
+use values::specified::image::{GradientKind, SizeKeyword};
```

Comment:
Specified `GradientKind` and `SizeKeyword` are not the same as computed. Lengths work differently in specified and computed contexts (that's almost the only difference between most specified/computed values 😄)

You'll need to mirror their definitions here